### PR TITLE
Migrate s2i docker tests to aos-cd-jobs

### PR DESCRIPTION
**Pull request by adambkaplan**
/assign @stevekuznetsov

### CODE REVIEW

File: sjb/config/test_cases/test_pull_requests_s2i.yml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+      wget -q https://dl.google.com/go/${BUILD_GO_VERSION}.linux-amd64.tar.gz && sudo tar -C /usr/local -xzf ${BUILD_GO_VERSION}.linux-amd64.tar.gz && sudo rm /bin/go* && sudo ln -s /usr/local/go/bin/go /usr/local/go/bin/godoc /usr/local/go/bin/gofmt /bin
+      make build
+      sudo cp /data/src/github.com/openshift/source-to-image/_output/local/bin/linux/amd64/s2i /usr/bin
+      echo "Printing s2i version"
+      s2i version
+  - type: script
+    title: build test images
+    repository: source-to-image
+    script: |-
+      hack/build-test-images.sh
```
Comment (adambkaplan) [inline sjb/config/test_cases/test_pull_requests_s2i.yml]: Turns out I didn't need the pull secret to registry.redhat.io after all (though having it available on ci.openshift.redhat.com isn't a bad thing - someone will use the UBI for their images sooner or later)

**Comment by adambkaplan** [comment]
fyi @openshift/openshift-team-developer-experience

File: sjb/config/test_cases/test_pull_requests_s2i.yml (hunk 1 above)
Comment (gabemontero) [reply sjb/config/test_cases/test_pull_requests_s2i.yml]: cool deal .... and refresh my memory, with https://github.com/openshift/release/blob/master/ci-operator/jobs/openshift/source-to-image/openshift-source-to-image-master-presubmits.yaml#L132-L141 you were able to confirm recently that running both prow based and jenkins based tests out of the same PR was functional, right?

In theory I could employ a similar pattern with https://github.com/openshift/aos-cd-jobs/blob/master/sjb/config/test_cases/test_branch_jenkins_images.yml to bring back those old docker/s2i based ginkgo tests

Comment (adambkaplan) [reply sjb/config/test_cases/test_pull_requests_s2i.yml]: > and refresh my memory, with https://github.com/openshift/release/blob/master/ci-operator/jobs/openshift/source-to-image/openshift-source-to-image-master-presubmits.yaml#L132-L141 you were able to confirm recently that running both prow based and jenkins based tests out of the same PR was functional, right?

Yes. Caveat here is that you need to hand-edit the `jobs/*.yaml` file - the config generator won't let you add `jenkins` agent tests.

**Comment by adambkaplan** [comment]
/hold

Going to rename the test so we don't tear down the old s2i jobs prematurely.

**Comment by adambkaplan** [comment]
/hold cancel

ping @openshift/openshift-team-developer-productivity-platform

**Comment by gabemontero** [comment]
bump  @openshift/openshift-team-developer-productivity-platform

**Comment by coverprice** [comment]
I think you meant to ping @openshift/openshift-team-developer-productivity-test-platform

**Comment by adambkaplan** [comment]
Pushed up to ci.openshift.redhat.com

**Pull request merged**